scripts/build_automation/suggest-lint-fix.swift

```diff
@@ -0,0 +1,196 @@
+
+ExternalCommand.verbose = true
+
+let repoOwner = "instructure"
+let repoName = "canvas-ios"
+let repo = "\(repoOwner)/\(repoName)"
+
+let env = ProcessInfo.processInfo.environment
+
+func envError(_ name: String) -> Never {
+    print("environment variable \(name) must be set")
+    exit(1)
+}
+
+enum Github {
+    static var token: String {
+        if let token = env["DANGER_GITHUB_API_TOKEN"] {
+            return token
+        }
+        if let token = env["GITHUB_ACCESS_TOKEN"] {
+            return token
+        }
+        envError("DANGER_GITHUB_API_TOKEN")
+    }
+
+    struct GraphQLRequest<C: Codable>: Codable {
+        let query: String
+        let variables: [String: C]
+    }
+
+    struct AddPullRequestReviewInput: Codable {
+        let threads: [DraftPullRequestReviewThread]
+        let commitOID: String
+        let event: PullRequestReviewEvent
+        let pullRequestId: String // global id, not the PR number
+        let body: String
+    }
+
+    enum PullRequestReviewEvent: String, Codable {
+        case approve = "APPROVE"
+        case comment = "COMMENT"
+        case dismiss = "DISMISS"
+        case requestChanges = "REQUEST_CHANGES"
+    }
+
+    struct DraftPullRequestReviewThread: Codable {
+        let body: String
+        // let commit_id: String
+        let path: String
+
+        let line: Int // last line
+
+        let side: String = "RIGHT"
+    }
+
+    static func findPullRequestId(prNumber: Int) throws -> String {
+        let query = """
+            query findPullRequestId($prNumber: Int!) {
+                repository(owner: "instructure", name: "canvas-ios") {
+                    pullRequest(number: $prNumber) { id }
+                }
+            }
+            """
+        return try (cmd(
+            "curl", "-sf", "https://api.github.com/graphql",
+            "-X", "POST",
+            "-H", "Authorization: Bearer \(token)",
+            "--data-binary", "@-"
+        ).inputJSON(from: GraphQLRequest(
+            query: query,
+            variables: [ "prNumber": prNumber ]
+        )) | cmd("jq", ".data.repository.pullRequest.id")).runJson(String.self)
+    }
+
+    static func postReview(_ input: AddPullRequestReviewInput) throws {
+        let query = """
+            mutation review($input: AddPullRequestReviewInput!) {
+                addPullRequestReview(input: $input) {
+                    pullRequestReview { url }
+                }
+            }
+            """
+        try (cmd(
+            "curl", "-sf", "https://api.github.com/graphql",
+            "-X", "POST",
+            "-H", "Accept: application/vnd.github.comfort-fade-preview+json",
+            "-H", "Authorization: Bearer \(token)",
+            "--data-binary", "@-"
+        ).inputJSON(from: GraphQLRequest(
+            query: query,
+            variables: [ "input": input ]
+        )) | cmd("jq")).run()
+    }
+}
+
+func inDir<R>(_ path: String, thunk: () throws -> R) rethrows -> R {
```

Comment:
Just a knit pick, seems like handler or completionHandler fits more with the swift paradigm than thunk